lib/engine/step/g_18_ms/track.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require_relative '../track'
+
+module Engine
+  module Step
+    module G18MS
+      class Track < Track
+        ACTION_HEX_OF_INTEREST = 'B10'
+        BONUS_HEX = 'B12'
+
+        def lay_tile(action, extra_cost: 0, entity: nil)
+          entity ||= action.entity
+          super
+
+          return if action.tile.hex.name != ACTION_HEX_OF_INTEREST ||
+            @game.chattanooga_reached ||
+            (!@game.loading && @game.graph.reachable_hexes(entity).find { |h, _| h.name == 'B12' }.nil?)
```

Comment:
still needs to be fixed

Reply:
I might not have understood. Have made yet another attempt.

What I do want to accomplish is to check if the tile placed in B10 creates a connection for the entity towards B12 (Chattanooga). So is there a better way to do this?

Reply:
the reason why you see game.loading checks is because any graph call (reachable hexes etc) are very expensive to compute. but anytime you see a game.loading check, it's purely to determine if something was legal or not, it doesn't have any side effects. however in your case, based on the graph call, you make a side effect.

that means that you cannot have different behavior for when game.loading and when game is not loading. thus, you should get rid of game.loading and pay the cost. or come up with something more clever.

Reply:
Ok, got it. In this case I think the cost is OK, as it just checked when placing tiles in one hex on the board (B10) that can trigger it, and it is very likely that they do connect to B12 when they do tile B10. 
So I have removed the game.load, and put the flag check first for readability.